pkg/identity/cache/allocator.go

```diff
@@ -482,3 +486,43 @@ func (m *CachingIdentityAllocator) RemoveRemoteIdentities(name string) {
 		m.IdentityAllocator.RemoveRemoteKVStore(name)
 	}
 }
+
+type IdentityChangeKind string
+
+const (
+	IdentityChangeSync   IdentityChangeKind = IdentityChangeKind(allocator.AllocatorChangeSync)
+	IdentityChangeUpsert IdentityChangeKind = IdentityChangeKind(allocator.AllocatorChangeUpsert)
+	IdentityChangeDelete IdentityChangeKind = IdentityChangeKind(allocator.AllocatorChangeDelete)
+)
+
+type IdentityChange struct {
+	Kind   IdentityChangeKind
+	ID     identity.NumericIdentity
+	Labels map[string]string
```

Comment:
Any particular reason not to use the native types for this?

```suggestion
	Labels labels.Labels
```

Reply:
Not that i'm aware of. Thanks for the input - I refactored it to use `labels.Labels`